src/tally.F90

```diff
@@ -3109,6 +3125,735 @@ subroutine score_surface_current(p)
   end subroutine score_surface_current
 
 !===============================================================================
+! APPLY_DERIVATIVE_TO_SCORE multiply the given score by its relative derivative
+!===============================================================================
+
+  subroutine apply_derivative_to_score(p, t, i_nuclide, atom_density, &
+                                       score_bin, score)
```

Comment:
This subroutine is a little sparse on comments. I like what you did below with `score_track_derivative` and `score_collision_derivative`. If you could do something similar here to give the code-reader a sense of where the formulas come from, I think it would be valuable.